packages/central-server/src/apiV2/GETMapOverlayGroupRelationChildren.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { BESAdminGETHandler } from './GETHandler';
+import { generateLinkHeader } from './GETHandler/helpers';
+
+export class GETMapOverlayGroupRelationChildren extends BESAdminGETHandler {
+  permissionsFilteredInternally = true;
+
+  async buildResponse() {
+    const options = await this.getDbQueryOptions();
+
+    // handle request for a single record
+    // const { recordId } = this;
+    // if (recordId) {
+    //   const record = await this.findSingleRecord(recordId, options);
+    //   return { body: record };
+    // }
+
+    // handle request for multiple records, including pagination headers
+
+    const criteria = await this.getDbQueryCriteria();
+    if (Object.keys(criteria).includes('childCode')) {
+      criteria.code = criteria.childCode;
+      delete criteria.childCode;
+    }
+    console.log('criteria', criteria);
+    const mapOverlayRecords = await this.findRecords('map_overlay', criteria, options);
+    const mapOverlayGroupRecords = await this.findRecords('map_overlay_group', criteria, options);
+    console.log('map overlay records', mapOverlayRecords);
+    console.log('map overlay records', mapOverlayGroupRecords);
+    const pageOfRecords = mapOverlayRecords
+      .map(o => {
+        return { childId: o.id, childCode: o.code, childType: 'Map Overlay' };
+      })
+      .concat(
+        mapOverlayGroupRecords
+          .filter(o => o.code !== 'Root') // exclude Root group from children list
+          .map(o => {
+            return { childId: o.id, childCode: o.code, childType: 'Map Overlay Group' };
+          }),
+      );
```

Comment:
I created this new endpoint due to the fact that all map overlay group relation child options are not easily retrieved with mapOverlayGroupRelation, as they are sourced from two separate tables and cannot be joined with map_overlay_group_relation, since not all children options would be already recorded in map_overlay_group_relation. I'm sure there are probably more logical strategies for this, but this is what I've come up with so far.